src/index.js

```diff
@@ -132,11 +132,15 @@ async function start(fields, doRetry = true) {
   } catch (err) {
     if (
       err.message === 'No refresh token is set.' ||
-      err.message === 'Invalid Credentials'
+      err.message === 'Invalid Credentials' ||
+      err.message === 'Request had insufficient authentication scopes.'
     ) {
-      if (!fields.refresh_token) {
+      if (err.message === 'Request had insufficient authentication scopes.') {
```

Comment:
There is no specific error code?

Reply:
it's a `401` error. It seems to be a `401` in those 3 cases:

```      err.message === 'No refresh token is set.' ||
      err.message === 'Invalid Credentials' ||
      err.message === 'Request had insufficient authentication scopes.'```

So we could do `if (err.code === 401)` instead

Reply:
Well in fact, I think it's better to use `401` and `403`